WalletWasabi/Helpers/ByteHelpers.cs

```diff
@@ -106,7 +106,7 @@ private static uint[] CreateLookup32Unsafe()
 		/// <summary>
 		/// Fastest byte array to hex implementation in C#
 		/// </summary>
-		public static string ToHex(params byte[] bytes)
+		public static string? ToHex(params byte[] bytes)
```

Comment:
I would say it would be better to get rid of the inner if statement:

```csharp
if (hex is null)
{
  return null;
}
```

so that one would use the function like this:

* ByteHelpers.ToHex(null) // this would be no longer valid
* ByteHelpers.ToHex() // returns empty string
* ByteHelpers.ToHex(0x00, ...) // returns non-empty string

Actually, it would be great to have unit tests containing:

```csharp
			Assert.Equal(null, ByteHelpers.ToHex(null));
			Assert.Equal("", ByteHelpers.ToHex(Array.Empty<byte>()));
            Assert.Equal("0102", ByteHelpers.ToHex(0x01, 0x02));
```

Reply:
> Assert.Equal(null, ByteHelpers.ToHex(null));

This will give a NRE.

Reply:
> This will give a NRE if we get rid of the inner if statement.

Yeah, as I said:

> ByteHelpers.ToHex(null) // this would be no longer valid

This `ByteHelpers.ToHex(null)` should be in a different PR if you decide to work on that.